source/logging.c

```diff
@@ -381,6 +381,17 @@ void aws_register_log_subject_info_list(struct aws_log_subject_info_list *log_su
     const uint32_t min_range = log_subject_list->subject_list[0].subject_id;
     const uint32_t slot_index = min_range >> AWS_LOG_SUBJECT_STRIDE_BITS;
 
+#if DEBUG_BUILD
```

Comment:
Actually, would this be better done in a test, so it will fail the build in CI?

Reply:
doing it in the register() function means that it will catch the problem in any other repo's subjects, and not just aws-c-common's log subjects 

I'd be fine taking out the `if DEBUG` and doing this in release too

Reply:
keeping it as is
we chatted offline
this approach keeps a linear scan out of RELEASE, but catches the issue if anyone runs anything in DEBUG